doc/guides/zephyr_cmake_package.rst

```diff
@@ -0,0 +1,418 @@
+.. _cmake_pkg:
+
+Zephyr CMake Package
+####################
+
+Zephyr can be exported as a CMake config package.
```

Comment:
As someone who doesn't know CMake packages, I didn't understand this intro section at all :P. It got better when I read the whole document, but if you want to make the introduction more noob-friendly, you could for example:
 - Explicitly define the target audience.
 - Assume people know nothing about CMake packages and the package registry going into this.
 - Give a bit more recommendations, when you know people will run into trouble if they don't do a certain thing.

Reply:
Hope the extended intro section is better